torch/_dynamo/output_graph.py

```diff
@@ -458,7 +473,29 @@ def wrap_name(module_key):
         base = name
         for i in itertools.count():
             if name not in self.nn_modules:
+                assert self.nn_modules_sources is not None
+                assert name not in self.nn_modules_sources
                 self.nn_modules[name] = target
+                self.nn_modules_sources[name] = source
+                if isinstance(target, torch.nn.Module):
+                    # annoying, but there are cases when we do not have parameters
+                    # see test_nn_moduledict_contains
```

Comment:
I don't understand this comment

Reply:
```
 class M(torch.nn.Module):
            def __init__(self, module_dict):
                super().__init__()
                self.module_dict = module_dict

            def forward(self, x):
                if "foo" in self.module_dict:
                    x = torch.mul(x, 1.0)
                x = torch.add(x, 1.0)
                return x

        module_dict = torch.nn.ModuleDict({"foo": torch.nn.Conv2d(1, 1, 1)})
        m = M(module_dict)
```

Is valid, but breaks some behavior and leads to a None `._parameters`

Reply:
@albanD @jbschlosser do you know about this? It seems weird to me that there exist modules for which `named_parameter` will fail if you call it on them.